### ISSUE TITLE
(v2) token kycStatus and freezeStatus are wrong way round

### ISSUE THREAD

**Issue by gregscullard** [hashgraph/hedera-sdk-js#292]
SDK reports KYC false when it should be true and vice-versa, same for freezeStatus. See test below
(Note: Tested the same with the Java SDK which works ok suggesting the issue is with the SDK and not services).

Expanded token transfer example to getAccountInfo and output relationships

account id = 0.0.1519
token id = 0.0.1520

created Token with KYC KEY, FREEZE KEY and default Freeze to "false" (per example).
Added accountinfo query to output result of each transaction

Associated account 0.0.1519 with token 0.0.1520
TokenRelationshipMap {
  _map: Map(1) {
    '0.0.1520' => TokenRelationship {
      tokenId: [TokenId],
      symbol: 'F',
      balance: [Long],
      kycStatus: true,
      freezeStatus: true
    }
  }
}
=> Account shows kycStatus=true, freezeStatus=true (should be false in both cases)

Granted KYC for account 0.0.1519 on token 0.0.1520
TokenRelationshipMap {
  _map: Map(1) {
    '0.0.1520' => TokenRelationship {
      tokenId: [TokenId],
      symbol: 'F',
      balance: [Long],
      kycStatus: false,
      freezeStatus: true
    }
  }
}
=> Account shows kycStatus=false, freezeStatus=true (kycStatus should be true)

Revoked KYC for account 0.0.1519 on token 0.0.1520
TokenRelationshipMap {
  _map: Map(1) {
    '0.0.1520' => TokenRelationship {
      tokenId: [TokenId],
      symbol: 'F',
      balance: [Long],
      kycStatus: true,
      freezeStatus: true
    }
  }
}
=> Account shows kycStatus=true, freezeStatus=true (kycStatus should be false)

**Comment by gregscullard** [comment]
Note, grant and revoke operations work as expected. After revocation, transfers are rejected by the network.

**Comment by gregscullard** [comment]
Maybe because kyc and freeze statuses aren't straight booleans ?

```
    enum TokenFreezeStatus {
        FreezeNotApplicable = 0,
        Frozen = 1,
        Unfrozen = 2
    }

    /** TokenKycStatus enum. */
    enum TokenKycStatus {
        KycNotApplicable = 0,
        Granted = 1,
        Revoked = 2
    }
```

**Comment by mehcode** [comment]
The intent was that they were nullable booleans: `NULL` = `0`, `true` = 1, and `false` = 2.

However, the property names make that pretty confusing. 

I vote for either using actual enums or renaming them so the booleans make sense:
 - `kycStatus` -> `isKycGranted`
 - `freezeStatus` -> `isFrozen`

**Comment by gregscullard** [comment]
For Frozen: Not sure the "not applicable" is necessary, you're either frozen or not (if it's not applicable, you're not frozen).
For KYC: not applicable kind makes sense, semantically it means something different to "Granted"

So maybe a mix of both.
-freezeStatus = boolean
-kycStatus=enum

**Comment by mehcode** [comment]
Works for me. @danielakhterov,

1. Rename `freezeStatus` to `isFrozen` and fix the boolean mismatch as explained in the opening comment.

2. Add a `KycStaus` enum with 3 variants to map to the protobuf.

**Comment by gregscullard** [comment]
@danielakhterov @mehcode thinking about it, the "not applicable" freeze status may be useful for user interfaces (it not applicable, it's disabled), saves additional logic checking for freeze keys, etc...
Would it be better to have:
-FreezeStatus: Enum (per Protobuf)
-isFrozen: Boolean (calculated)
and maybe
-KycStatus: Enum (per Protobuf)
-isKYCd: Boolean (calculated)

**Comment by danielakhterov** [comment]
@gregscullard The issue here was that I was checking the wrong number resulting in inverted values for `kycStatus` and `freezeStatus`. Perhaps what we should have is simply rename `kycStatus` to `isKycGranted: boolean | null` and `freezeStatus` to `isFrozen: boolean | null`. The 3 state boolean would encompass the enum while also being fairly readable. If `isFrozen` is true then account is frozen; if false account is not frozen; if null being frozen is not applicable on this token. UI would also be able to use this to disable buttons if needed.

**Comment by gregscullard** [comment]
That works too.

**Comment by danielakhterov** [comment]
Updated and fixed in v2.0.1
